modules/core/src/main/java/org/apache/ignite/configuration/DefragmentationConfiguration.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.configuration;
+
+import java.io.Serializable;
+
+/**
+ *
+ */
+public final class DefragmentationConfiguration implements Serializable {
```

Comment:
Do we need a dedicated `DefragmentationConfiguration`? Can we store all these properties in `DataStorageConfiguration`?

Reply:
Yes, we can, it's a subject for discussion I guess.

Reply:
I suggest minimizing changes. If the wider defragmentation configuration scope appears we can create a such a class configuration, but currently, I see no reasons.

Reply:
We'll try to derive region sizes from current configuration. Not exactly the same way as @agoncharuk described, but approach will be similar. I hope that it won't take much time.